sdcm/cluster.py

```diff
@@ -4385,6 +4389,34 @@ def stop_nemesis(self, timeout=10):
                 stack_trace = traceback.format_stack(current_thread_frames[nemesis_thread.ident])
                 threads_tracebacks.append("\n".join(stack_trace))
 
+    def start_kms_key_rotation_thread(self, rotation_interval_m=60) -> None:
+        if self.params.get("cluster_backend") != 'aws':
+            return None
+        kms_key_alias_name = ""
+        append_scylla_yaml = yaml.safe_load(self.params.get("append_scylla_yaml")) or {}
+        for kms_host_name, kms_host_data in append_scylla_yaml.get("kms_hosts", {}).items():
+            if "auto" in kms_host_name:
+                kms_key_alias_name = kms_host_data["master_key"]
+                break
+        if not kms_key_alias_name:
+            self.log.debug("Automatic KMS key is not configured. Skip creation of it's rotation thread.")
+            return None
+
+        def _rotate_kms_key(kms_key_alias_name, rotation_interval_m):
```

Comment:
any thread we should decorate with `@raise_event_on_failure` otherwise we won't notice failures on it

Reply:
Ok, will do.